### PR TITLE
RND203-326: ar9271Probe implementation

### PR DESCRIPTION
**Pull request by Rafal-Litwinski-Mobica**
if_ath_usb_devlist.h splitted in two files (if_ath_usb_devid.h) to easy reuse vendor and products IDs.

### CODE REVIEW

File: sys/dev/ath/ath_hal/ar9002/ar9271_attach.c [hunk 1]
```diff
@@ -493,12 +495,37 @@ ar9271FillCapabilityInfo(struct ath_hal *ah)
...
-	// if (vendorid == ATHEROS_VENDOR_ID && (devid == AR2427_DEVID_PCIE))
-	// 	return "Atheros 2427";
-
+	if (vendorid == USB_VENDOR_ATHEROS2) {
+		if (devid == USB_PRODUCT_ATHEROS2_AR9271_1 ||
+			devid == USB_PRODUCT_ATHEROS2_AR9271_2 ||
+			devid == USB_PRODUCT_ATHEROS2_AR9271_3)
+			return "Atheros 9271";
+	}
+	if (vendorid == USB_VENDOR_AZUREWAVE) {
```
Comment (Mikolaj-Filar-Mobica) [inline sys/dev/ath/ath_hal/ar9002/ar9271_attach.c:504]: `if` => `else if` or `switch`?

File: sys/dev/ath/ath_hal/ar9002/ar9271_attach.c [hunk 2]
```diff
@@ -493,12 +495,37 @@ ar9271FillCapabilityInfo(struct ath_hal *ah)
...
 {
-	//TODO: add vendor AR9271 vendor
-	// if (vendorid == ATHEROS_VENDOR_ID && devid == AR9285_DEVID_PCIE)
-	// 	return "Atheros 9285";
-	// if (vendorid == ATHEROS_VENDOR_ID && (devid == AR2427_DEVID_PCIE))
-	// 	return "Atheros 2427";
-
+	if (vendorid == USB_VENDOR_ATHEROS2) {
+		if (devid == USB_PRODUCT_ATHEROS2_AR9271_1 ||
+			devid == USB_PRODUCT_ATHEROS2_AR9271_2 ||
```
Comment (Mikolaj-Filar-Mobica) [inline sys/dev/ath/ath_hal/ar9002/ar9271_attach.c]: Here and in other places: 
- alignment
- I'm not sure if syntax with additional brackets for complex expressions is preferred i.e.: 
         
        if ((devid == USB_PRODUCT_ATHEROS2_AR9271_1) ||
	    (devid == USB_PRODUCT_ATHEROS2_AR9271_2) ||

File: sys/dev/ath/if_ath_usb_devlist.h [hunk 3]
```diff
@@ -20,41 +20,7 @@
  */
```
Comment (Mikolaj-Filar-Mobica) [inline sys/dev/ath/if_ath_usb_devlist.h:21]: It's not part of this commit but I see this file has no protection for multiple/recursive inclusions: `#ifndef __IF_ATH_USB_DEVLIST_H__`

Comment (Rafal-Litwinski-Mobica) [reply sys/dev/ath/if_ath_usb_devlist.h:21]: This files was copy from original sys/dev/ath/if_ath_usb_devlist.h doesn't have such protection. I was wondering if I should fix it, but finally it was done in the same way as original file.

File: sys/dev/ath/ath_hal/ar9002/ar9271_attach.c (hunk 2 above)
Comment (Rafal-Litwinski-Mobica) [reply sys/dev/ath/ath_hal/ar9002/ar9271_attach.c]: I will fix it.

File: sys/dev/ath/ath_hal/ar9002/ar9271_attach.c (hunk 1 above)
Comment (Rafal-Litwinski-Mobica) [reply sys/dev/ath/ath_hal/ar9002/ar9271_attach.c:504]: It's in the same way like other ar92xxProbe, please check i.e. ar9287Probe.

File: sys/dev/ath/if_ath_usb_devlist.h (hunk 3 above)
Comment (Rafal-Litwinski-Mobica) [reply sys/dev/ath/if_ath_usb_devlist.h:21]: Fixed

File: sys/dev/ath/ath_hal/ar9002/ar9271_attach.c (hunk 2 above)
Comment (Rafal-Litwinski-Mobica) [reply sys/dev/ath/ath_hal/ar9002/ar9271_attach.c]: Done

**Comment by Mikolaj-Filar-Mobica** [comment]
I see that alignment is done with tabs & spaces, e.g.:
image

**Review by Mikolaj-Filar-Mobica — APPROVED** [review]
